sources/OpenMcdf/CompoundFile.cs

```diff
@@ -523,7 +523,7 @@ public void Commit(bool releaseMemory)
                 CheckForLockSector();
 
             sourceStream.Seek(0, SeekOrigin.Begin);
-            sourceStream.Write((byte[])Array.CreateInstance(typeof(byte), GetSectorSize()), 0, sSize);
+            sourceStream.Write(new byte[sSize], 0, sSize);

```

Comment:
I also changed the array size from ```GetSectorSize()``` to ```sSize``` (the ```sSize``` variable is initialized to ```GetSectorSize()``` above, but I didn't think it made sense for the call here to create the array with ```GetSectorSize``` size and then call ```Write``` with ```sSize``` as the size?